File: backend/src/main/java/com/marketinganalytics/platform/entity/Lead.java
package com.marketinganalytics.platform.entity;

import com.marketinganalytics.platform.entity.enums.LeadSource;
import com.marketinganalytics.platform.entity.enums.LeadStatus;
import jakarta.persistence.*;
import lombok.AllArgsConstructor;
import lombok.Builder;
import lombok.Getter;
import lombok.NoArgsConstructor;
import lombok.Setter;

import java.time.Instant;

@Entity
@Table(name = "leads")
@Getter
@Setter
@NoArgsConstructor
@AllArgsConstructor
@Builder
public class Lead {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;

    @Column(nullable = false, length = 150)
    private String name;

    @Column(nullable = false, length = 180)
    private String email;

    @Column(length = 30)
    private String phone;

    @Column(length = 150)
    private String company;

    @Enumerated(EnumType.STRING)
    @Column(nullable = false, length = 20)
    private LeadSource source;

    @Enumerated(EnumType.STRING)
    @Column(nullable = false, length = 20)
    @Builder.Default
    private LeadStatus status = LeadStatus.NEW;

    @Column(nullable = false)
    @Builder.Default
    private int score = 0;

    @Column(name = "contact_count", nullable = false)
    @Builder.Default
    private int contactCount = 0;

    @ManyToOne(fetch = FetchType.LAZY)
    @JoinColumn(name = "campaign_id")
    private Campaign campaign;

    @ManyToOne(fetch = FetchType.LAZY)
    @JoinColumn(name = "assigned_to")
    private User assignedTo;

    @Column(columnDefinition = "TEXT")
    private String notes;

    @Column(name = "created_at", nullable = false, updatable = false)
    private Instant createdAt;

    @Column(name = "updated_at", nullable = false)
    private Instant updatedAt;

    @PrePersist
    void onCreate() {
        Instant now = Instant.now();
        createdAt = now;
        updatedAt = now;
    }

    @PreUpdate
    void onUpdate() {
        updatedAt = Instant.now();
    }
}

File: frontend/src/types/lead.ts
import type { LeadSource, LeadStatus, LeadTemperature } from './common'

export interface Lead {
  id: number
  name: string
  email: string
  phone: string | null
  company: string | null
  source: LeadSource
  status: LeadStatus
  score: number
  contactCount: number
  temperature: LeadTemperature
  campaignId: number | null
  campaignName: string | null
  assignedToId: number | null
  assignedToName: string | null
  notes: string | null
  createdAt: string
  updatedAt: string
}

export interface LeadRequest {
  name: string
  email: string
  phone?: string
  company?: string
  source: LeadSource
  campaignId?: number | null
  assignedToId?: number | null
  notes?: string
}
